### PR TITLE
fix: Improve getFileExtension() readability and handle leading dot extensions.

### PR DESCRIPTION
**Pull request by damanV5**
## Description
More robust edge case handling for getFileExtension().

**Case** : Type is not provided explicitly in {src,type} object.
In this case getFileExtension() is called to fetch the extension so as to get the type.

For certain cases, fn does not returns the correct extension.
For Eg: 
1. https://demo.unified-streaming.com/k8s/features/stable/video/tears-of-steel/tears-of-steel.ism/.m3u8

In this case fn returns "". 
I have simplified getFileExtension() so that fn will return m3u8 in this case as well.

## Specific Changes proposed
Simplified getFileExtension() for better readability and to handle more edge cases.

## Requirements Checklist
- [x] Feature implemented / Bug fixed
- [ ] If necessary, more likely in a feature request than a bug fix
  - [x] Change has been verified in an actual browser (Chrome, Firefox, IE)
  - [x] Unit Tests updated or fixed
  - [ ] Docs/guides updated
  - [ ] Example created ([starter template on JSBin](https://codepen.io/gkatsev/pen/GwZegv?editors=1000#0))
  - [x] Has no DOM changes which impact accessiblilty or trigger warnings (e.g. Chrome issues tab)
  - [x] Has no changes to JSDoc which cause `npm run docs:api` to error
- [x] Reviewed by Two Core Contributors

### CODE REVIEW

**Comment by damanV5** [comment]
@gkatsev Could you please help with getting this merged?

**Review by mister-ben — DISMISSED** [review]
Thanks for the PR! It's a nice simplification. It would consider `com` in `https://example.com` a file extensions too, but so does the original.

Could you add a test to test/until/utils/url.test.js with the type of URL that should now work?

I don't think "accept a training slash" is the best description. When I read that I thought it's intended for a URL like `.../file.m3u8/`. Something like to "match an extension without a filename"?

**Comment by damanV5** [comment]
@mister-ben  I have added test cases in `test/until/utils/url.test.js` and updated the PR description as well.
Let me know if something else needs to be done.

File: test/unit/utils/url.test.js [hunk 1]
```diff
@@ -55,6 +55,8 @@ QUnit.test('should get the file extension of the passed path', function(assert)
   assert.equal(Url.getFileExtension('foo/.bar/test.video.flv?foo=bar'), 'flv');
   assert.equal(Url.getFileExtension('http://www.test.com/video.mp4'), 'mp4');
   assert.equal(Url.getFileExtension('http://foo/bar/test.video.wgg'), 'wgg');
+  assert.equal(Url.getFileExtension('http://www.test.com/video/.mp4'), 'mp4');
+  assert.equal(Url.getFileExtension('http://www.test.com/video/.mp4/'), 'mp4');
```
[the remark this answers is not in the document]
Comment (damanV5) [reply test/unit/utils/url.test.js:59]: @mister-ben The updated regex will also handle a trailing slash , which was not handled before.

Though, I can remove it.

[the remark this answers is not in the document]
Comment (damanV5) [reply test/unit/utils/url.test.js:59]: I just checked the original regex also handled the trailing slash, but there was no test case written for that.

[the remark this answers is not in the document]
Comment (damanV5) [reply test/unit/utils/url.test.js:59]: @mister-ben

[the remark this answers is not in the document]
Comment (alex-barstow) [reply test/unit/utils/url.test.js:59]: @damanV5 Thx for the contribution! Would you mind adding an additional test case that has a complete file name + extension + trailing slash? ex. `'http://www.test.com/video.mp4/'`

In addition to the case of having only an extension (`'/.mp4'`), are there other specific known edge cases this change will handle? If not, then I'd recommend changing the PR title to something more specific than "handle edge cases", for example: `fix: Improve getFileExtension() readability and handle leading dot extensions`

**Review by alex-barstow — CHANGES_REQUESTED** [review]
A couple very small recommendations but otherwise LGTM! Thx @damanV5

File: test/unit/utils/url.test.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (damanV5) [reply test/unit/utils/url.test.js:59]: Sure @alex-barstow , Let me add that.

**Comment by damanV5** [comment]
@alex-barstow  I have checked, this enhancement will handle the following test cases :-

 `http://www.test.com/video/.mp4`
`http://www.test.com/video/.mp4/`


-----

`http://www.test.com/video.mp4/` : This was already handled but test case was not added , so I have now added a test case.

File: test/unit/utils/url.test.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (damanV5) [reply test/unit/utils/url.test.js:59]: I have added `http://www.test.com/video.mp4/` in url.test.js.

**Comment by damanV5** [comment]
@alex-barstow  Sure. I have now updated the PR title and updated the base branch.

**Review by alex-barstow — APPROVED** [review]
Approved. Just need one more contributor to approve

**Comment by damanV5** [comment]
@gkatsev  / @heff  Can you please approve this?

**Review by adrums86 — APPROVED** [review]
Thanks for your contribution! LGTM.
